toucan_connectors/soap/soap_connector.py

```diff
@@ -73,11 +74,27 @@ def _retrieve_data(self, data_source: SoapDataSource) -> pd.DataFrame:
         # Force the casting of response
         response = getattr(client.service, data_source.method)(**data_source.parameters)
 
-        if isinstance(response, Sequence):
-            result = pd.DataFrame(response)
+        if is_list_response(response):
+            if is_nested_list(response):
+                result = (
+                    pd.DataFrame(response[0]) if len(response[0]) > 0 else pd.DataFrame(None)
```

Comment:
```suggestion
                    pd.DataFrame(response[0]) if len(response) > 0 else pd.DataFrame(None)
```
plutot ? sinon ton `len(response[0])` va te faire un indexError aussi

Sinon j'ai l'impression que ce serait plus simple de mettre le teste `len(response) > 0` dans tes utils `is_nested_list` et `is_dict_of_lists` 
Ton cas else retombera directement sur une DatFrame vide (équivalent à `else: pd.DataFrame([])` )

Reply:
merci, remarque prise en compte.